src/core/server/http/logging/get_payload_size.ts

```diff
@@ -25,9 +26,23 @@ const isFsReadStream = (src: unknown, res: Response): src is ReadStream => {
     res.source instanceof ReadStream
   );
 };
+const isZlibStream = (src: unknown, res: Response): src is Zlib => {
+  return (
+    !isBoom(res) &&
+    res.variety === 'stream' &&
+    res.source === src &&
```

Comment:
Why we need such a sophisticated check? Maybe duck-typing would be enough?

Reply:
Re-evaluating each of the typeguards, there's actually quite a bit of cleanup we can do -- for example, we're already checking `isBoom` earlier in the function so we don't need those checks duplicated in the typeguards, and testing reference equality of `res.source` and `src` isn't strictly necessary.

I went ahead and simplified each of them to take the `source` object and the `variety` string.

The one for zlib streams is still a little more complex as TS will complain if we just try to check `variety === 'stream' && obj.bytesWritten`, as `obj` is `unknown`, hence the `typeof` and `null` checks.